### PR TITLE
feat: implement search context and UA based on selenium 4.21

### PR DESCRIPTION
**Pull request by KazuCocoa**
Aims to implement search context and UA based on https://github.com/SeleniumHQ/selenium/pull/12141

### CODE REVIEW

File: lib/appium_lib_core/common/base/http_default.rb [hunk 1]
```diff
@@ -39,6 +42,16 @@ def initialize(open_timeout: nil, read_timeout: nil)
...
           end
 
+          def set_additional_header(key, value)
+            @additional_headers[key] = value
+            # ::Selenium::WebDriver::Remote::Http::Common.extra_headers = @additional_headers
+          end
+
+          def delete_additional_header(key)
+            @additional_headers.delete key
+            # ::Selenium::WebDriver::Remote::Http::Common.extra_headers = @additional_headers
```
Comment (KazuCocoa) [inline lib/appium_lib_core/common/base/http_default.rb:52]: this kept adding headers added by `set_additional_header` so currently the implementation keeps L76.

**Comment by KazuCocoa** [comment]
ok, anyway this implementation works.

File: lib/appium_lib_core/common/base/http_default.rb (hunk 1 above)
Comment (p0deje) [reply lib/appium_lib_core/common/base/http_default.rb:52]: Is this something to fix on Selenium side?

Comment (KazuCocoa) [reply lib/appium_lib_core/common/base/http_default.rb]: Hm, maybe no? I noticed that needed to remove from `@header` since we have one time header only for a new session creation. Instead of adding `Common.extra_headers`, making `@headers` accessible is also enough as our usage https://github.com/appium/ruby_lib_core/pull/480/commits/34afcb9ddbcf7e56bb4c4385da61278ea8cb8f8d

File: lib/appium_lib_core/common/device/context.rb [hunk 2]
```diff
@@ -37,10 +37,10 @@ def current_context
...
           end
 
-          def available_contexts
-            # return empty array instead of nil on failure
-            execute(:available_contexts, {}) || []
-          end
+          # def available_contexts
+          #   # return empty array instead of nil on failure
+          #   execute(:available_contexts, {}) || []
+          # end
```
Comment (KazuCocoa) [inline lib/appium_lib_core/common/device/context.rb:43]: this comment out is to test add_command

File: lib/appium_lib_core/common/base/bridge.rb [hunk 3]
```diff
@@ -158,6 +166,18 @@ def json_create(value)
...
+        # Example:
+        #   driver.add_command(name: :available_contexts, method: :get, url: 'session/:session_id/contexts') do
+        #     execute(:available_contexts, {}) || []
+        #   end
+        # Then,
+        #   driver.available_contexts #=> ["NATIVE_APP"]
+
+        # def add_command(method:, url:, name:, &block)
+        #   Bridge.add_command name, method, url, &block
+        # end
```
Comment (KazuCocoa) [inline lib/appium_lib_core/common/base/bridge.rb:179]: todo: once we move to this bridge.add_command, need to update tests as well. should be another pr

File: lib/appium_lib_core/common/base/bridge.rb [hunk 4]
```diff
@@ -231,52 +251,9 @@ def element_attribute(element, name)
...
         # For Appium
-        # override
-        def active_element
-          ::Appium::Core::Element.new self, element_id_from(execute(:get_active_element))
-        end
         alias switch_to_active_element active_element
 
-        # For Appium
-        # override
-        def find_element_by(how, what, parent_ref = [])
```
Comment (KazuCocoa) [inline lib/appium_lib_core/common/base/bridge.rb:242]: these simplification looks great
